### DISCUSSION TITLE
Static file not found: jquery-3.7.1.slim.js [docker] [1.30.4]

### DISCUSSION THREAD

**Discussion by working-name** [dani-garcia/vaultwarden#4399] [Q&A] [answered]
### Subject of the issue
Admin interface cannot load jquery-3.7.1.slim.js.

Possibly related to #3884 but I have not modified admin templates, and vaultwarden is running behind NPM (nginx proxy manager), and have toggled its cache on and off. I don't see any /tmp files either.

```json
{"ErrorModel":{"Message":"Static file not found: jquery-3.7.1.slim.js","Object":"error"},"ExceptionMessage":null,"ExceptionStackTrace":null,"InnerExceptionMessage":null,"Message":"Static file not found: jquery-3.7.1.slim.js","Object":"error","ValidationErrors":{"":["Static file not found: jquery-3.7.1.slim.js"]},"error":"","error_description":""}
```

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.30.4
* Web-vault version: v2024.1.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.44.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": false,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*****************",
  "domain_origin": "*****://*****************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "NoSaltWarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/vaultwarden.log",
  "log_level": "debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "push_enabled": true,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": true,
  "signups_verify_resend_limit": 10,
  "signups_verify_resend_time": 60,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "*********************",
  "smtp_from_name": "Vault Warden",
  "smtp_host": "****************",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "*********************",
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

* vaultwarden version: v1.30.4

* Install method: Docker

* Clients used: web vault

* Reverse proxy and version: NPM v2.10.4

* MySQL/MariaDB or PostgreSQL version: no

* Other relevant details: Upgraded from 1.30.0 I think, by doing `docker compose pull && docker compose up -d`

### Steps to reproduce
Upgrade to 1.30.4, login to admin, then open browser console. You'll notice the UI isn't responsive in parts where jQuery is used.

### Expected behaviour
Load jQuery.

### Actual behaviour
jQuery is not loaded.

### Troubleshooting data

```bash
$ docker compose exec vaultwarden curl localhost/vw_static/jquery-3.7.1.slim.js
{"ErrorModel":{"Message":"Static file not found: jquery-3.7.1.slim.js","Object":"error"},"ExceptionMessage":null,"ExceptionStackTrace":null,"InnerExceptionMessage":null,"Message":"Static file not found: jquery-3.7.1.slim.js","Object":"error","ValidationErrors":{"":["Static file not found: jquery-3.7.1.slim.js"]},"error":"","error_description":""}
```

**Comment by BlackDex** [comment]
You probably have either overwritten the static/template files, or are using some caching mechanisch.

**Reply by working-name** [reply]
Check out troubleshooting data - that's vaultwarden itself answering to a curl request running inside the container. Not a caching problem, unless `vw_static/jquery-3.7.1.slim.js` is not supposed to exist but my browser thinks so.

I have not touched any files, I've simply `tar zcf` to backup `./data` mount.

**Comment by dani-garcia** [comment, marked as answer]
Seems like this reference didn't get updated, could that be causing the issue?
https://github.com/dani-garcia/vaultwarden/blob/d6b97090fab5eba429a79c35735a0760b2fa9ee2/src/api/web.rs#L176-L177

**Reply by working-name** [reply]
That must be it. If I curl for `3.7.0` I get the long js file back. Same for the browser, which is requesting it via nginx proxy manager.

**Comment by THEDARKK** [comment]
Same problem

**Comment by BlackDex** [comment]
Fixed in testing now. We will release a patch soon.

**Comment by driverspy** [comment]
Thanks for the quick fix.
I couldn't delete, deactivate users or resend invitations in the admin dashboard.

It's been fixed since the update, thanks!

**Comment by Kruzerson** [comment]
Hello @dani-garcia. I upgraded to version **1.30.5** and still can't delete a user through the admin panel. Decatification and other actions are not available either.

**Reply by BlackDex** [reply]
Doubt it has something to do with this topic. 
What are the error messages you get either in the browser or in the Vaultwarden logs?

**Reply by Kruzerson** [reply]
![image](https://github.com/dani-garcia/vaultwarden/assets/130353475/ad1b8526-f7f7-412e-8b16-bc15bc2e0d2a)
